File: 240229/키, 몸무게를 기준으로 정렬/sort-by-height-and-weight.js
class Person {
    constructor(n, h, w) {
        this.n = n;
        this.h = h;
        this.w = w;
    }
}

const fs = require("fs");
let [n, ...inputs] = fs.readFileSync(0).toString().trim().split("\n");

const persons = []
for(const p of inputs) {
    const person = p.trim().split(" ")
    persons.push(new Person(person[0], person[1], person[2]))
}

function compare(a, b) {
    if(a.h === b.h) return b.w - a.w
    return a.h - b.h
}

persons.sort(compare)
for(const p of persons) {
    console.log(`${p.n} ${p.h} ${p.w}`)
}

File: 240229/키, 몸무게를 기준으로 정렬/sort-by-height-and-weight.py
# 키 - 오름차순
# 키 동일 - 몸무게 내림차순

class Person:
    def __init__(self, n, h, w):
        self.n, self.h, self.w = n, h, w

n = int(input())
persons = []

for _ in range(n):
    n,h,w = tuple(input().split())
    persons.append(Person(n,int(h),int(w)))

persons.sort(key=lambda x: (x.h, -x.w) )

for person in persons:
    print(person.n, person.h, person.w)